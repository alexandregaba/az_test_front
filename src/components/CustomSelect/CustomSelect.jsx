import React from 'react';
import styled from 'styled-components';

const CustomSelectContainer = styled.div`
  position: relative;
  font-family: Arial;
  width: 100%;
  margin-left: 10px;
  font-size: 1rem;

  &:after {
    position: absolute;
    content: '';
    top: ${props => (props.isOpened ? '4px' : '11px')};
    right: 10px;
    width: 0;
    height: 0;
    border: 6px solid transparent;
    border-color: ${props =>
      props.isOpened
        ? 'transparent transparent #fff transparent'
        : '#fff transparent transparent transparent'};
  }
`;

const OptionsList = styled.div`
  background-color: #dcdcdc;
`;

const SelectedOption = styled(OptionsList)`
  padding-left: 10px;
`;

const Option = styled.div`
  display: ${props => (props.show ? 'block' : 'none')};
  padding-left: 10px;
  border-bottom: solid 1px black;

  &:hover {
    background-color: #cdd0dd;
  }
`;

const CustomSelect = ({ options, selectMe, toggleSelectOpen, isOpened, selected, ...props }) => (
  <CustomSelectContainer isOpened={isOpened} {...props}>
    <SelectedOption onClick={toggleSelectOpen}>{selected.display}</SelectedOption>
    <OptionsList>
      {options.map((option, index) => (
        <Option show={isOpened} onClick={selectMe} key={index}>
          {option.display}
        </Option>
      ))}
    </OptionsList>
  </CustomSelectContainer>
);

export default CustomSelect;
