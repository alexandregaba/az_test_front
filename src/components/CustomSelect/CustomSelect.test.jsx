import React from 'react';
import CustomSelect from './CustomSelect';
import { shallow } from 'enzyme';

const props = {
  options: [
    { value: '', display: 'All', isDefault: true },
    { value: 'checked', display: 'Checked Only' },
    { value: 'notChecked', display: 'Not checked Only' },
  ],
  selectMe: jest.fn(),
  toggleSelectOpen: jest.fn(),
  isOpened: false,
  selected: { value: '', display: 'All', isDefault: true },
};
it('matches snapshot', () => {
  const ClosedComponent = shallow(<CustomSelect {...props} />);
  expect(ClosedComponent).toMatchSnapshot();

  const OpenComponent = shallow(<CustomSelect {...props} isOpened />);
  expect(OpenComponent).toMatchSnapshot();
});
