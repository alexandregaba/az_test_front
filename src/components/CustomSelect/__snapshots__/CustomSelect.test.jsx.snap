// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styled.div
  isOpened={false}
>
  <Styled(styled.div)
    onClick={[MockFunction]}
  >
    All
  </Styled(styled.div)>
  <styled.div>
    <styled.div
      key="0"
      onClick={[MockFunction]}
      show={false}
    >
      All
    </styled.div>
    <styled.div
      key="1"
      onClick={[MockFunction]}
      show={false}
    >
      Checked Only
    </styled.div>
    <styled.div
      key="2"
      onClick={[MockFunction]}
      show={false}
    >
      Not checked Only
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`matches snapshot 2`] = `
<styled.div
  isOpened={true}
>
  <Styled(styled.div)
    onClick={[MockFunction]}
  >
    All
  </Styled(styled.div)>
  <styled.div>
    <styled.div
      key="0"
      onClick={[MockFunction]}
      show={true}
    >
      All
    </styled.div>
    <styled.div
      key="1"
      onClick={[MockFunction]}
      show={true}
    >
      Checked Only
    </styled.div>
    <styled.div
      key="2"
      onClick={[MockFunction]}
      show={true}
    >
      Not checked Only
    </styled.div>
  </styled.div>
</styled.div>
`;
